src/python/grpcio_tests/tests/unit/_api_test.py

```diff
@@ -117,20 +118,32 @@ def test_secure_channel(self):
         channel = grpc.secure_channel("google.com:443", channel_credentials)
         channel.close()
 
-    def test_multiple_secure_channel(self):
+    @pytest.mark.parametrize("use_compute_engine_credentials_in_test", [False, True])
+    def test_multiple_secure_channel(use_compute_engine_credentials_in_test):
         _THREAD_COUNT = 10
         wait_group = test_common.WaitGroup(_THREAD_COUNT)
 
-        def create_secure_channel():
-            channel_credentials = grpc.ssl_channel_credentials()
+        def create_secure_channel(use_compute_engine_credentials):
+            if use_compute_engine_credentials:
+                class TestCallCredentials(grpc.AuthMetadataPlugin):
```

Comment:
Can we put this class outside of test functions?

Reply:
Fixed in 9c56890ce93943617a40b045f5a808b72ff4522b